TF_Service_and_Tooling_Unity/Assets/TouchFree/ServiceUI/Scripts/PermissionController.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Ultraleap.TouchFree.ServiceShared;
+
+public class PermissionController : MonoBehaviour
+{
+    public static bool hasFilePermission = true;
+
+    // Force a save which causes the check for file access and if we do not
+    // have it, sets hasFilePermission to false
+    private void Awake()
+    {
+        ConfigManager.PhysicalConfig.SaveConfig();
```

Comment:
It feels a bit strange that we set `hasFilePermission` via a side effect of the `SaveConfig()`  method. Could `SaveConfig` return a bool or have an `out` parameter that sets the value so how this value gets set is clearly defined from this class?

Reply:
I thought the same, but we also have other saves that do not need to know if it succeeded or not. I didn't think it was worth adding a specific function here to handle it and it happening upon every SaveConfig() feels like it's usefully double bagging it (also means we don't throw errors when it fails)

Returning a bool may be the nicest way to do this though to make it more explicit? I feel a out is overkill particularly when we don't always need to know the result

Reply:
I hadn't considered the benefit of this updating on every save. Given that I think this is fine as it is